### ISSUE TITLE
Student issue while trying to replicate your results...

### ISSUE THREAD

**Issue by FxbxxnSxhxxrxxxx** [Lihaogx/PAA#1]
Hey together :),

I am trying out your project / code, but I am stumbling upon a time out error that has to do with the proxy to my understanding... 
How did you solve that if that happened to you as well?

Thanks a lot already in advance for your response and this interesting paper to read :))

BG
Fabian

**Comment by Lihaogx** [comment]
Hi Fabian,

Thanks a lot for your interest in our project and for taking the time to try it out!

The timeout error you're encountering is likely due to the proxy settings that were for our local development.  
If you don’t need to use a proxy on your end, you can **simply delete or comment out** the following lines in both  
`PAA/agent/legislator_agent.py` and `PAA/agent/decision_pathway.py`:

```python
os.environ["http_proxy"] = "http://localhost:7890"
os.environ["https_proxy"] = "http://localhost:7890"
```
This should fix the issue. Let us know if anything else comes up, and thanks again for the kind words about the paper! 😊

BG,
Hao Li

**Comment by FxbxxnSxhxxrxxxx** [comment]
Yes :). It's working / processing now the profiles. 

Thanks a lot for this fast response! I am curious about the results when trying this on my own :).

**Comment by Lihaogx** [comment]
Great to hear that it's working now! 

Feel free to reach out if you run into any other issues or have ideas you'd like to discuss!

**Comment by FxbxxnSxhxxrxxxx** [comment]
> Great to hear that it's working now!
> 
> Feel free to reach out if you run into any other issues or have ideas you'd like to discuss!

As I am trying it out right now: Just to understand it better: 

1. Using GPT-4o-mini with your provided framework: Does this basically refer to "PAA_G" as in your paper? 
2. Does using the "20-profiles433" folder from this data folder as underlying training data refers to the mentioned 4/3/3 data split? 

![Image](https://github.com/user-attachments/assets/7c7438bb-a7df-4cdc-a48b-4503e448957d)

**Comment by FxbxxnSxhxxrxxxx** [comment]
Also I was wondering how you deal with prediction results like this:

![Image](https://github.com/user-attachments/assets/6dae3fbd-905c-4742-9747-6901b339984e)
![Image](https://github.com/user-attachments/assets/fd2f494a-bb93-416c-83b8-c27221ac83b2)

Because in some cases, I stumple upon predictions that are entire sentences, and instead of "Nay" or "Yea" I get back "Aye" or even predictions without any of these words...

**Comment by Lihaogx** [comment]
Hi Fabian,

1. Yes, using GPT-4o-mini with our framework corresponds to the **PAA\_G** variant described in the paper.

2. And yes again, using the `20-profiles433` folder refers to the **4/3/3 data split** setup we used in our experiments.

3. Regarding your observation about outputs like `"Aye"` or full sentences: that's expected. In our postprocessing, we treat `"Aye"` as equivalent to `"Yea"` and `"No"` as equivalent to `"Nay"`. If the model outputs a full sentence, we extract the final decision by searching for these keywords.

   To our knowledge, the model has always produced a choice in some form. But if you're seeing cases where no clear decision is made, we'd really appreciate it if you could share a few examples.

Thanks again for the detailed feedback!

Best regards,  
Hao Li

**Comment by FxbxxnSxhxxrxxxx** [comment]
Hey Hao,

sure, let me attach my jupyter notebook and the results here so far :).
There you can see four entries out of all other predictions (aprox. 8000) where I feel that I cannot really
retrieve a personalized prediction. 

Another question: 
What I was wondering as well is, maybe I did not understand it within the code, but how does the framework actually know that the split of 4/3/3 shall be done for example and not 6/2/2 using the other folder for example? Or is it encoded within the data of the profiles themselves?

Side note: I only run a sample from the profiles for the folder 20-profiles433.

[20-result433_run1.csv](https://github.com/user-attachments/files/20463757/20-result433_run1.csv)
[jupyter_notebook_as_pdf.pdf](https://github.com/user-attachments/files/20463830/jupyter_notebook_as_pdf.pdf)

Best regards,
Fabian

**Comment by Lihaogx** [comment]
Hi Fabian,

Thanks a lot for sharing your notebook and results.

To respond to your questions:

1. About the personalized prediction parsing: 
   You're absolutely right — in the vast majority of cases, the decision can be retrieved using simple keyword matching (e.g., "Yea", "Nay", "Aye", "No"). Only in very rare situations (like your 4 out of ~8000 cases) does it require manual classification. We appreciate you pointing those out!

2. About the 4/3/3 split logic
   Our framework itself doesn’t enforce any train/val/test split — there is no internal mechanism for splitting data dynamically (e.g., 4/3/3 vs 6/2/2). Instead, the split is encoded in the **data preprocessing stage**.  
   So for example, in the `20-profiles433` folder, the profiles were manually selected to reflect a 4/3/3 setup. Similarly, a `622` folder would contain profiles pre-selected based on a 60\%/20\%/20\% ratio. All data under the `data/` directory are preprocessed and fixed in this way.

Thanks again for the detailed follow-up!

Best regards,  
Hao

**Comment by FxbxxnSxhxxrxxxx** [comment]
Hey Hao,

I just run a short evaluation. Maybe you can make use of it :).
I added a distribution-based negative control as well.

Just an impression, but would you say that there is slight bias towards 'Yea'?
I just run a sample as mentioned before since I didn't have more credits for GPT so I don't know
if, in terms of the target variable, the splits are balanced or stay as imbalanced as I have them in my sample. 
It would be interesting to see how the framework performs if the target variable is fully balanced :)

Best regards,
Fabian

[documented_eval.pdf](https://github.com/user-attachments/files/20483904/documented_eval.pdf)

**Comment by Lihaogx** [comment]
Hi Fabian,

Thanks a lot for the follow-up and for sharing the evaluation — the negative control addition is really thoughtful!

To your question about a potential bias towards "Yea", here are a few thoughts:

1. You're absolutely right, the imbalance exists.
   While our framework doesn’t have a traditional training process (and technically no train/val/test split in the usual sense), the imbalance toward "Yea" is definitely present in the data.  
   This actually reflects the nature of real-world political processes, most bills that make it to the floor for a vote already have strong bipartisan or majority support, so "Yea" is more common in actual outcomes.

2. Some embedding-based baselines do involve training, but as far as we’ve seen, they don’t explicitly apply any label balancing operations.

3. We really like your suggestion about balancing the training data or adding more negative (i.e., "Nay") examples into the profiles.  
   That could be a valuable direction to test how robust the framework is under controlled label distributions. We’ll definitely try that in future runs and see how it impacts the results.

Thanks again for your thoughtful testing!

Best regards,  
Hao
